compiler/luci/pass/src/FuseMulToFullyConnectedWeightsPass.test.cpp

```diff
@@ -0,0 +1,160 @@
+
+  void init(void)
+  {
+    TestIOGraph::init({3, 4}, {3, 6});
+
+    _mul = g()->nodes()->create<luci::CircleMul>();
+    _mul_s = g()->nodes()->create<luci::CircleConst>();
+    _fc = g()->nodes()->create<luci::CircleFullyConnected>();
+    _fc_w = g()->nodes()->create<luci::CircleConst>();
+    _fc_b = g()->nodes()->create<luci::CircleConst>();
+
+    _mul->name("mul");
+    _mul_s->name("mul_s");
+    _fc->name("fc");
+    _fc_w->name("fc_w");
+    _fc_b->name("fc_b");
+
+    _mul->dtype(DT);
+    _fc->dtype(DT);
+    _mul->fusedActivationFunction(luci::FusedActFunc::NONE);
+    _fc->fusedActivationFunction(luci::FusedActFunc::NONE);
+
+    _mul_s->rank(1);
+    _mul_s->dim(0) = 3;
+    _mul_s->dtype(DT);
+    _mul_s->size<DT>(3);
+    for (uint32_t i = 0; i < 3; ++i)
+    {
+      _mul_s->at<DT>(0) = 1.0f;
+    }
+
+    _fc_w->rank(2);
+    _fc_w->dim(0) = 3;
+    _fc_w->dim(1) = 4;
+    _fc_w->dtype(DT);
+    _fc_w->size<DT>(4 * 6);
+    for (uint32_t i = 0; i < 4 * 6; ++i)
+    {
+      _fc_w->at<DT>(0) = 1.0f;
+    }
+
+    _fc_b->rank(1);
+    _fc_b->dim(0) = 6;
+    _fc_b->dtype(DT);
+    _fc_b->size<DT>(6);
+    for (uint32_t i = 0; i < 6; ++i)
+    {
+      _fc_b->at<DT>(0) = 1.0f;
+    }
+
+    _mul->x(input());
+    _mul->y(_mul_s);
+    _fc->input(_mul);
+    _fc->weights(_fc_b);
+    _fc->bias(_fc_b);
+
+    output()->from(_fc);
+  }
+
+  luci::CircleMul *_mul = nullptr;
+  luci::CircleFullyConnected *_fc = nullptr;
+  luci::CircleConst *_mul_s = nullptr;
+  luci::CircleConst *_fc_w = nullptr;
+  luci::CircleConst *_fc_b = nullptr;
+};
+
+class FuseMulToFullyConnectedWeightsPassTest : public ::testing::Test
+{
+public:
+  FuseMulToFullyConnectedWeightsPassTest() = default;
+
+protected:
+  FuseMulToFullyConnectedWeightsPassTestGraph<loco::DataType::FLOAT32> _graph;
+  luci::FuseMulToFullyConnectedWeightsPass _pass;
+};
+
+class FuseMulToFullyConnectedWeightsPassS32Test : public ::testing::Test
+{
+public:
+  FuseMulToFullyConnectedWeightsPassS32Test() = default;
+
+protected:
+  FuseMulToFullyConnectedWeightsPassTestGraph<loco::DataType::S32> _graph;
+  luci::FuseMulToFullyConnectedWeightsPass _pass;
+};
+
+} // namespace
+
+TEST_F(FuseMulToFullyConnectedWeightsPassTest, name)
+{
+  auto const name = _pass.name();
+  ASSERT_NE(nullptr, name);
+}
+
+TEST_F(FuseMulToFullyConnectedWeightsPassTest, fuse_mul_to_fc_weights)
+{
+  _graph.init();
+
+  EXPECT_TRUE(_pass.run(_graph.g()));
+}
+
+TEST_F(FuseMulToFullyConnectedWeightsPassTest, mul_fused_act_NEG)
+{
+  _graph.init();
+
+  _graph._mul->fusedActivationFunction(luci::FusedActFunc::RELU);
+
+  EXPECT_FALSE(_pass.run(_graph.g()));
+}
+
+TEST_F(FuseMulToFullyConnectedWeightsPassTest, mul_d2_NEG)
+{
+  _graph.init();
+
+  _graph._mul_s->rank(2);
+  _graph._mul_s->dim(0) = 1;
+  _graph._mul_s->dim(1) = 3;
+
+  EXPECT_FALSE(_pass.run(_graph.g()));
+}
+
+TEST_F(FuseMulToFullyConnectedWeightsPassS32Test, dtype_s32_NEG)
+{
+  _graph.init();
+
+  EXPECT_FALSE(_pass.run(_graph.g()));
+}
```

Comment:
This test seems same with the `fuse_mul_to_fc_weights` test, but it expects `false`.
How could it be..!?

Reply:
`FuseMulToFullyConnectedWeightsPassS32Test` is made with S32, which datatype we can't support.

Reply:
Ah, it was `S32`.
Thank you!